ext/libxml/ruby_xml_node.c

```diff
@@ -45,13 +45,18 @@ static void rxml_node_deregisterNode(xmlNodePtr xnode)
   /* Has the node been wrapped and exposed to Ruby? */
   if (xnode->_private)
   {
+    /* Does it belong to another libxml user? */
+    if (!OWNED(xnode))
```

Comment:
It's entirely possible. I was not 100% sure if we wanted to also be cleaning up regardless of the `xnode->_private` check being performed and solely rely on the `OWNED(xnode)` check. I was not  sure what other conditions may be introduced if we removed this.